spot_driver/src/robot_state/spot_robot_state_middleware_handle.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver/robot_state/spot_robot_state_middleware_handle.hpp>
+
+namespace {
+constexpr auto kPublisherHistoryDepth = 1;
+constexpr auto kNodeName{"spot_robot_state_publisher"};
+
+// ROS topic names for Spot's robot state publisher
+constexpr auto kJointStatesTopic{"joint_states"};
+constexpr auto kOdomTwistTopic{"odometry/twist"};
+constexpr auto kOdomTopic{"odometry"};
+constexpr auto kFeetTopic{"status/feet"};
+constexpr auto kEStopTopic{"status/estop"};
+constexpr auto kWifiTopic{"status/wifi"};
+constexpr auto kBatteryStatesTopic{"status/battery_states"};
+constexpr auto kPowerStatesTopic{"status/power_states"};
+constexpr auto kSystemFaultsTopic{"status/system_faults"};
+constexpr auto kBehaviorFaultsTopic{"status/behavior_faults"};
+constexpr auto kEndEffectorForceTopic{"status/end_effector_force"};
+constexpr auto kManipulatorTopic{"manipulation_state"};
+}  // namespace
+
+namespace spot_ros2 {
+
+SpotRobotStateMiddlewareHandle::SpotRobotStateMiddlewareHandle(const std::shared_ptr<rclcpp::Node>& node)
+    : node_{node},
+      parameter_interface_{std::make_unique<RclcppParameterInterface>(node)},
+      logger_interface_{std::make_unique<RclcppLoggerInterface>(node->get_logger())},
```

Comment:
The same applies to the `logger_interface` and the `timer_interface`. The `tf_interface` can stay as it is used by the class.

Reply:
Refactored to move these around.

I decided to move `tf_interface` out as well, since it simplifies constructing these objects if it's at the same level as the middleware handle.